Tutorial.playground/Contents.swift

```diff
@@ -0,0 +1,484 @@
+let fromTwoToSix_ = { $0 + 1 } <^> fromOnetoFive
+
+fromTwoToSix_.generate
+fromTwoToSix_.generate
+fromTwoToSix_.generate
+
+let generatorBoundedSizeArrays_ = fromOnetoFive >>- { len in
+	return characterArray.suchThat { xs in xs.count <= len }
+}
+
+generatorBoundedSizeArrays_.generate
+generatorBoundedSizeArrays_.generate
+generatorBoundedSizeArrays_.generate
+
+//: Now that you've seen what generators can do, we'll use all we've learned to create a generator that
+//: produces email addresses.  To do this, we'll need one more operator/method notated `<*>` or `ap`.
+//: `ap` comes from [Applicative Functors](http://staff.city.ac.uk/~ross/papers/Applicative.html) and is
+//: used to "zip together" `Gen`erators of functions with `Gen`erators of of values, applying each function
+//: during the zipping phase.  That definition is a little hand-wavey and technical, so for now we'll say that
+//: `ap` works like "glue" that sticks special kinds of generators togethers.
+//:
+//: For our purposes, we will say that an email address consists of 3 parts: A local part, a hostname, and a 
+//: Top-Level Domain each separated by an `@`, and a `.` respectively.
+//:
+//: According to RFC 2822, the local part can consist of uppercase characters, lowercase letters, numbers, and
+//: certain kinds of special characters.  We already have generators for upper and lower cased letters, so all we
+//: need are special characters and a more complete number generator:
+
+let numeric : Gen<Character> = Gen<Character>.fromElementsIn("0"..."9")
+let special : Gen<Character> = Gen<Character>.fromElementsOf(["!", "#", "$", "%", "&", "'", "*", "+", "-", "/", "=", "?", "^", "_", "`", "{", "|", "}", "~", "."])
+
+//: Now for the actual generator
+
+let allowedLocalCharacters : Gen<Character> = Gen<Character>.oneOf([
+	upperCaseLetters,
+	lowerCaseLetters,
+	numeric,
+	special,
+])
+
+//: Now we need a `String` made of these characters. so we'll just `proliferate` an array of characters and `fmap`
+//: to get a `String` back.
+
+let localEmail = allowedLocalCharacters.proliferateNonEmpty().fmap(String.init)
+
+//: The RFC says that the host name can only consist of lowercase letters, numbers, and dashes.  We'll skip some
+//: steps here and combine both steps above into one big generator.
+
+let hostname = Gen<Character>.oneOf([
+	lowerCaseLetters,
+	numeric,
+	Gen.pure("-"),
+]).proliferateNonEmpty().fmap(String.init)
+
+//: Finally, the RFC says the TLD for the address can only consist of lowercase letters with a length larger than 1.
+
+let tld = lowerCaseLetters.proliferateNonEmpty().suchThat({ $0[$0.endIndex.predecessor()] != "." }).fmap(String.init)
+
+//: So now we've got all the pieces together, so how do we put them together to make the final generator?  Well, how
+//: about some glue?
+
+// Concatenates 5 strings together in order.
+func glue5(l : String) -> String -> String -> String -> String -> String {
```

Comment:
This should work too:

    func glue5(l: String)(m: String)(m2: String)(m3: String)(r: String) {
        return l + m + m2 + m3 + r
    }